folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+    if (
+      !this.localeTranslations.containsKey(configuration.getDefaultLocale())
+    ) {
+      this.localeTranslations.put(
+        configuration.getDefaultLocale(),
+        this.resolveDefaultLocale()
+      );
+    }
+    return this.localeTranslations.get(configuration.getDefaultLocale());
+  }
+
+  /**
+   * Get all of the locales for the current request.  This will respect the Accept-Language header
+   * and quality values, as applicable, with highest quality first.
+   *
+   * @return a List of all Locale objects.  Invalid and wildcard Locales will likely return a Locale
+   *   with empty string/fields or, in some cases, the server's current Locale.
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language">MDN docs</a>
+   */
+  @SuppressWarnings("java:S1166")
+  public List<Locale> getCurrentLocales() {
+    try {
+      return Collections.list(
+        (
+          (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()
+        ).getRequest()
+          .getLocales()
+      );
+    } catch (IllegalStateException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Get the best single locale.  Either the first from the request or the server's default.
+   *
+   * @return a single Locale object.  Invalid and wildcard Locales sent to the server will likely
+   *   return a Locale with empty string/fields or, in some cases, the server's current Locale.
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language">MDN docs</a>
+   */
+  @SuppressWarnings("java:S1166")
+  public Locale getCurrentLocale() {
+    try {
+      return getCurrentLocales().get(0);
+    } catch (RuntimeException e) {
+      return configuration.getDefaultLocale();
+    }
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} for the current language(s).
+   * This will respect the Accept-Language header and quality values, as applicable.
+   *
+   * @return the best TranslationMap
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language"> MDN docs</a>
+   */
+  public TranslationMap getCurrentTranslation() {
+    return this.getBestTranslation(this.getCurrentLocales());
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} for the provides list of Locales.
+   * This will return the first one that matches on at least language or, if none match,
+   * the default TranslationMap.
+   *
+   * @param locales the ordered list of locales to consider
+   * @return the best TranslationMap
+   */
+  public TranslationMap getBestTranslation(Iterable<Locale> locales) {
+    // return the first one that is a good match
+    for (Locale locale : locales) {
+      TranslationMap correspondingMap = this.getTranslation(locale);
+      if (correspondingMap.getQuality() != TranslationMatchQuality.NO_MATCH) {
+        return correspondingMap;
+      }
+    }
+
+    return this.getDefaultTranslation();
```

Comment:
This feels like it's just begging to be a stream

```suggestion
   return locales.stream()
                 .map(this::getTranslation)
                 .filter(m -> m.getQuality() != TranslationMatchQuality.NO_MATCH)
                 .findFirst()
                 .orElseGet(this::getDefaultTranslation);

```